test/getters.test.js

```diff
@@ -0,0 +1,300 @@
+import { jDataView } from "../src/jdataview";
+import {
+	compareBytes,
+	compareWithNaN,
+	getPrefilledJDataView,
+	b,
+	chr,
+	bufferToHex,
+} from "./test-helpers";
+import { describe, test, assert } from "vitest";
+
+describe("Getters", function () {
+	// getter = value || {value, check?, view?, args?}
+	function testGetters(type, getters) {
+		const view = getPrefilledJDataView();
+		test(type, function () {
+			getters.forEach(function (getter) {
+				if (typeof getter !== "object") {
+					getter = { value: getter };
+				}
+
+				const args = getter.args || [];
+				const contextView = getter.view || view;
+				const check = getter.check || assert.equal;
+				const value = getter.value;
+				const offset = contextView.tell();
+				const realValue = contextView["get" + type].apply(
+					contextView,
+					args
+				);
+
+				check(
+					realValue,
+					value,
+					"get" +
+						type +
+						"(" +
+						args.join(", ") +
+						") == " +
+						realValue +
+						" != " +
+						value +
+						" at offset " +
+						offset +
+						(getter.view
+							? " in view [" +
+							  getter.view
+									.getBytes(undefined, 0, true)
+									.join(", ") +
+							  "]"
+							: "")
+				);
+			});
+		});
+	}
+
+	testGetters("Bytes", [
+		{ args: [2, 1], value: [0xfe, 0xfd], check: compareBytes },

```

Comment:
I guess if we just removed the `compareBytes` helper and turned `value` here into `Uint8Array.of(0xfe, 0xfd)`, it would catch those type mismatches?

Reply:
Or idk change `compareBytes` to auto-convert argument via `Uint8Array.from`. Either one is fine.

Reply:
`compareBytes` is used in quite a few places, so I went with the latter.